### PR TITLE
Remove old http-proxy, use the full version as the official http-proxy example

### PR DESCRIPTION
**Pull request by vmalloc**

### CODE REVIEW

File: http-proxy/Cargo.toml [hunk 1]
```diff
@@ -1,21 +1,15 @@
 [package]
-name = "http-proxy"
+name = "http-full-proxy"
```
Comment (JohnTitor) [inline http-proxy/Cargo.toml]: IMO: The package, workspace, and directory name should have the same one. What about?

Comment (svenstaro) [reply http-proxy/Cargo.toml]: Sorry, what do you mean?

Comment (svenstaro) [reply http-proxy/Cargo.toml]: Oh I see, yes. It should be named `http-full-proxy`.

Comment (vmalloc) [reply http-proxy/Cargo.toml]: Oops, missed that. Fixed

File: http-proxy/Cargo.toml [hunk 2]
```diff
@@ -1,21 +1,15 @@
 [package]
-name = "http-proxy"
+name = "http-full"
```
Comment (JohnTitor) [inline http-proxy/Cargo.toml]: Ah, the name is still invalid.
```suggestion
name = "http-proxy"
```

Comment (vmalloc) [reply http-proxy/Cargo.toml]: Oops, I feel stupid now. Fixed!

**Comment by vmalloc** [comment]
I restored the SSL feature dependency because it should be able to use https, of course...

File: http-proxy/Cargo.toml [hunk 3]
```diff
@@ -1,21 +1,14 @@
 [package]
 name = "http-proxy"
 version = "0.1.0"
-authors = ["Nikolay Kim <fafhrd91@gmail.com>"]
-edition = "2018"
+authors = ["Rotem Yaari"]
```
Comment (leizzer) [inline http-proxy/Cargo.toml]: is it needed to change the author? In any case why not listing both? (real question, don't know how the community handle crate authors)

Comment (Dowwie) [reply http-proxy/Cargo.toml]: I've authored several examples and haven't bothered adding myself to the list.  Nikolay has maintained the repo after initial submissions are made.   An original author plus Nikolay seems fair.

Thoughts?

Comment (vmalloc) [reply http-proxy/Cargo.toml]: Sure, I will add myself with Nikolai going first of course. The reason for this is that I just moved the full http-proxy example that I wrote from scratch over to this new one...

**Pull request merged**